Installing package with period in name (normalized to a hyphen) fails on 8.1.2

- Pip version: 8.1.2
- Python version: 2.7.9
- Operating System: OS X 10.11.5
### Description:

Package `github3.py` has a name that should be normalized to `github3-py` and is when you view it on pypi.io. If I then tell pip to install the hyphenated name, I get an exception.
### What I've run:

```
$ pip install -i https://pypi.io/simple github3-py
```

This results in an installed package but an error:

```
Exception:
Traceback (most recent call last):
  File "/Users/sigmavirus24/.test/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Users/sigmavirus24/.test/lib/python2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/Users/sigmavirus24/.test/lib/python2.7/site-packages/pip/req/req_set.py", line 742, in install
    **kwargs
  File "/Users/sigmavirus24/.test/lib/python2.7/site-packages/pip/req/req_install.py", line 831, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/Users/sigmavirus24/.test/lib/python2.7/site-packages/pip/req/req_install.py", line 1032, in move_wheel_files
    isolated=self.isolated,
  File "/Users/sigmavirus24/.test/lib/python2.7/site-packages/pip/wheel.py", line 348, in move_wheel_files
    assert info_dir, "%s .dist-info directory not found" % req
AssertionError: github3-py .dist-info directory not found
```

If I then try using `github3.py` I see that the package is installed:

```
$ pip install -i https://pypi.io/simple github3.py
Requirement already satisfied (use --upgrade to upgrade): github3.py in ./.test/lib/python2.7/site-packages
Requirement already satisfied (use --upgrade to upgrade): uritemplate.py>=0.2.0 in ./.test/lib/python2.7/site-packages (from github3.py)
Requirement already satisfied (use --upgrade to upgrade): requests>=2.0 in ./.test/lib/python2.7/site-packages (from github3.py)
```

--- a/pip/wheel.py
+++ b/pip/wheel.py
@@ -298,9 +298,8 @@ def move_wheel_files(name, req, wheeldir, user=False, home=None, root=None,
                     continue
                 elif (is_base and
                         s.endswith('.dist-info') and
-                        # is self.req.project_name case preserving?
-                        s.lower().startswith(
-                            req.name.replace('-', '_').lower())):
+                        canonicalize_name(s).startswith(
+                            canonicalize_name(req.name))):
                     assert not info_dir, ('Multiple .dist-info directories: ' +
                                           destsubdir + ', ' +
                                           ', '.join(info_dir))

